### ISSUE TITLE
Version 4.0.0 (book/repo) is outdated/broken

### ISSUE THREAD

**Issue by jdickey** [TalkingQuickly/rails-server-template#64]
After all, a year and nine months is a long time in open-source software with lots of independent moving parts. That (and this issue) isn't meant as a dig on the author; it's just a fact of software life.

It starts as little things, like outdated versions breaking `berks vendor`, but [this issue fix](https://github.com/TalkingQuickly/rails-server-template/issues/63#issuecomment-370830446) works and you can go on; no biggie.

Then you get to the point in Chapter 5 where you run `knife zero converge`, and [all hell breaks loose](https://gist.github.com/jdickey/732deb848aa6cebf0df5549f726f4a57).

I rather suspect that `ohai` has changed something important and that's the cause of the breakage. I further suspect that I am like many of the buyers of the book: completely new to Chef; the only O**j**ai I'd previously heard of is a town in California; under geologically-significant pressure to get deployment automation up and happy to end an embarrassingly-long flail.

Help?

**Comment by danielristic** [comment]
While the book and repo has been –and still is– a great resource to create production-grade environments and deploy rails apps to them, it is getting increasingly difficult to end up with a stable system due to many moving parts as you pointed out, and an update of some of the oldest dependencies is long overdue.

As it is now, readers who strictly follow the instructions in the book will most likely end up with some kind of expected errors in the process, or if they're lucky enough to reach the end of the `knife zero converge` command, with a functioning but only partially provisioned server (example [here](https://github.com/TalkingQuickly/rails-server-template/issues/59)).

In your specific case,  I see you followed the [fix](https://github.com/TalkingQuickly/rails-server-template/issues/63#issuecomment-370830446) I suggested a while back but if you look at my latest entry on the thread, I discarded that and suggested keeping the outdated versions of the cookbooks and instead make sure you install Chef 12 and not a more recent version that is incompatible.

Looking at the output you provided, I see that chef 14.5.33 was used which most likely responsible of the error. However, note that you are also trying to provision an Ubuntu 18.04 system while the book has only been tested on 16.04 LTS, and even if you downgrade chef to version 12, you might encouter a different error.
